envbuilder.go

```diff
@@ -902,6 +838,340 @@ ENTRYPOINT [%q]`, exePath, exePath, exePath)
 	return nil
 }
 
+// RunCacheProbe performs a 'dry-run' build of the image and checks that
+// all of the resulting layers are present in options.CacheRepo.
+// Logger is the logf to use for all operations.
+// Filesystem is the filesystem to use for all operations.
+// Defaults to the host filesystem.
+func RunCacheProbe(ctx context.Context, opts options.Options) (v1.Image, error) {
+	if !opts.GetCachedImage {
+		return nil, fmt.Errorf("developer error: RunCacheProbe must be run with --get-cached-image")
+	}
+	if opts.CacheRepo == "" {
+		return nil, fmt.Errorf("--cache-repo must be set when using --get-cached-image")
+	}
+
+	stageNumber := 0
+	startStage := func(format string, args ...any) func(format string, args ...any) {
+		now := time.Now()
+		stageNumber++
```

Comment:
Nit: pass in an int pointer instead perhaps?

Reply:
This is a pre-existing non-pointer integer; I want to refactor this whole stage builder business later on. A little copying for now.